program/src/state/rules.rs

```diff
@@ -183,13 +194,27 @@ impl Rule {
         match self {
             Rule::All { .. } => 0,
             Rule::Any { .. } => 1,
-            Rule::AdditionalSigner { .. } => 2,
-            Rule::PubkeyMatch { .. } => 3,
-            Rule::DerivedKeyMatch { .. } => 4,
-            Rule::ProgramOwned { .. } => 5,
-            Rule::Amount { .. } => 6,
-            Rule::Frequency { .. } => 7,
-            Rule::PubkeyTreeMatch { .. } => 8,
+            Rule::Not { .. } => 2,
+            Rule::AdditionalSigner { .. } => 3,
+            Rule::PubkeyMatch { .. } => 4,
+            Rule::DerivedKeyMatch { .. } => 5,
+            Rule::ProgramOwned { .. } => 6,
+            Rule::Amount { .. } => 7,
+            Rule::Frequency { .. } => 8,
+            Rule::PubkeyTreeMatch { .. } => 9,
+        }
+    }
```

Comment:
Do we still need this particular function?  I didn't think it was used any more.